### ISSUE TITLE
Gradle multi module project with Kotlin 1.8.21 and `java-test-fixtures` plugin fail to test/build

### ISSUE THREAD

**Issue by andrezimmermann** [quarkusio/quarkus#34061]
### Describe the bug

After upgrading project from Quarkus 3.0.4.Final to Quarkus 3.1.1.Final, when running `./gradlew build`, tests started failing with with `jakarta.enterprise.inject.AmbiguousResolutionException`

The project is a Gradle multi module project, using Kotlin. Some of the sub-modules to use `java-test-fixtures` to share common test fixtures. 

I've created a reproducer and pushed to a public GitHub repository.

### Expected behavior

When updating the Quarkus application to 3.1.1.Final, the previous behavior of 3.0.4.Final would be kept, allowing to run tests with a submodule that uses the `java-test-fixtures`

Running `./gradlew build` would result in a successful build

### Actual behavior

Running `./gradlew build` will result in error

```

> Task :app:test

  2023-06-15 13:33:21,257 WARN  [io.qua.arc.dep.SplitPackageProcessor] (build-26) Detected a split package usage which is considered a bad practice and should be avoided. Following packages were detected in multiple archives: 
  - "org.acme.dao" found in [5c7507f26c01e1d823c326aa3ccf009a5288ae08:app-1.0.0-SNAPSHOT::jar, org.acme:app::jar, org.acme:lib::jar]
  - "org.acme" found in [5c7507f26c01e1d823c326aa3ccf009a5288ae08:app-1.0.0-SNAPSHOT::jar, org.acme:app::jar]

org.acme.GreetingResourceTest

  Test testHelloEndpoint() FAILED

  java.lang.RuntimeException: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
Error:   	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: Found 2 deployment problems: 
  [1] Ambiguous dependencies for type org.acme.dao.GreetingService and qualifiers [@Default]
  	- java member: org.acme.GreetingResource#service
  	- declared on CLASS bean [types=[org.acme.GreetingResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.GreetingResource]
  	- available beans:
  		- CLASS bean [types=[java.lang.Object, org.acme.dao.GreetingService], qualifiers=[@Default, @Any], target=org.acme.dao.GreetingService]
  		- CLASS bean [types=[java.lang.Object, org.acme.dao.GreetingService], qualifiers=[@Default, @Any], target=org.acme.dao.GreetingService]
  [2] Ambiguous dependencies for type org.acme.dao.GreetingService and qualifiers [@Default]
  	- java member: org.acme.GreetingResource#service
  	- declared on CLASS bean [types=[org.acme.GreetingResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.GreetingResource]
  	- available beans:
  		- CLASS bean [types=[java.lang.Object, org.acme.dao.GreetingService], qualifiers=[@Default, @Any], target=org.acme.dao.GreetingService]
  		- CLASS bean [types=[java.lang.Object, org.acme.dao.GreetingService], qualifiers=[@Default, @Any], target=org.acme.dao.GreetingService]
```

Refer here for the GitHub Actions logs (https://github.com/andrezimmermann/quarkus-kotlin-with-test-fixture/actions/runs/5279449037/jobs/9550135035)

### How to Reproduce?

Reproducer: 

I've pushed a small project that imitates the structure of my project, with Quarkus 3.1.1.Final

1. clone https://github.com/andrezimmermann/quarkus-kotlin-with-test-fixture
2. run `./gradlew build`, this would fail

I've added a branch with Quarkus 3.0.4.Final, that successfully builds  
1. clone https://github.com/andrezimmermann/quarkus-kotlin-with-test-fixture/tree/passing-build
2. run `./gradlew build`, this would pass

### Output of `uname -a` or `ver`

Linux desktop 6.3.7-x64v3-xanmod1 #0~20230609.ad1a983 SMP PREEMPT_DYNAMIC Sat Jun 10 00:20:29 UTC  x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.5" 2022-10-18 OpenJDK Runtime Environment GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08) OpenJDK 64-Bit Server VM GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08, mixed mode, sharing)

### Quarkus version or git rev

3.1.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.1.1

### Additional information

With Quarkus 3.1.1.Final, setting the kotlin version to 1.8.10, asides from the Kotlin daemon compilation errors, does successfully builds.

```
> Task :app:compileTestKotlin
Failed to compile with Kotlin daemon: java.lang.NoSuchMethodError: 'void org.jetbrains.kotlin.daemon.common.IncrementalCompilationOptions.<init>(boolean, java.util.List, java.util.List, org.jetbrains.kotlin.incremental.ClasspathChanges, java.io.File, org.jetbrains.kotlin.daemon.common.CompilerMode, org.jetbrains.kotlin.daemon.common.CompileService$TargetPlatform, java.lang.Integer[], int, java.lang.Integer[], boolean, java.util.List, org.jetbrains.kotlin.daemon.common.MultiModuleICSettings, org.jetbrains.kotlin.incremental.IncrementalModuleInfo, java.lang.String[], boolean)'
```

That is why initially I thought was a Kotlin issue, after investigating it further, I came to the conclusion that was indeed `java-test-fixtures` plugin that does not play well with Kotlin version. Maybe is directly related with the usage of the Kotlin daemon for compilation?

Removing the `java-test-fixtures` does makes the Quarkus 3.1.1.Final and Kotlin 1.8.21 build pass, which indicates is the plugin that interferes with some changes from Quarkus 3.1.1.Final. (https://github.com/andrezimmermann/quarkus-kotlin-with-test-fixture/pull/2)

Sorry for the trouble, any help would be appreciated

**Comment by geoand** [comment]
cc @mkouba

**Comment by aloubyansky** [comment]
We need to check the application model initialization in this case.

**Comment by glefloch** [comment]
Looks like a dependency is detected twice in the model no?

**Comment by andrezimmermann** [comment]
After investigating some more, this was caused by Kotlin 1.8.20 and fixed at Kotlin 1.8.22.
See: https://youtrack.jetbrains.com/issue/KT-58280

Trying out with 1.8.22 and Quarkus 3.2.0.CR1 at the reproducer project I created, the issue reported resolves itself. 
This can be closed (not sure if I can just close it).
Sorry for the trouble

**Comment by geoand** [comment]
Thanks for the update, that's very useful information!
